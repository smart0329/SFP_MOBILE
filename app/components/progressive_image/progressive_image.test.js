// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ProgressiveImage from './progressive_image';

jest.mock('react-native-fast-image', () => ({
    preload: jest.fn(),
}));

jest.useFakeTimers();

describe('ProgressiveImage', () => {
    const baseProps = {
        isBackgroundImage: false,
        defaultSource: 0,
        filename: 'defaultImage',
        imageUri: 'https://images.com/image.png',
        onError: jest.fn(),
        resizeMethod: 'auto',
        resizeMode: 'contain',
        theme: Preferences.THEMES.default,
        thumbnailUri: 'https://images.com/image.png',
        tintDefaultSource: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<ProgressiveImage {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should load image', () => {
        const wrapper = shallow(<ProgressiveImage {...baseProps}/>);
        const instance = wrapper.instance();
        jest.spyOn(instance, 'load');

        // should load image on componentDidMount
        instance.componentDidMount();
        expect(instance.load).toHaveBeenCalledTimes(1);

        // should not re-load image on componentDidUpdate with same props
        instance.componentDidUpdate(baseProps);
        expect(instance.load).toHaveBeenCalledTimes(1);

        // should re-load image on componentDidUpdate when props changed
        wrapper.setProps({filename: 'newImage'});
        expect(instance.load).toHaveBeenCalledTimes(2);

        wrapper.setProps({imageUri: 'https://images.com/new_image.png'});
        expect(instance.load).toHaveBeenCalledTimes(3);

        wrapper.setProps({thumbnailUri: 'https://images.com/new_image.png'});
        expect(instance.load).toHaveBeenCalledTimes(4);
    });

    test('should set image when imageUri is set but not the thumbnailUri', () => {
        const wrapper = shallow(
            <ProgressiveImage
                {...baseProps}
                thumbnailUri={null}
            />,
        );
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setImage');

        instance.componentDidMount();
        jest.runAllTimers();
        expect(instance.setImage).toHaveBeenCalledTimes(1);
    });

    test('should set thumbnail when thumbnailUri and imageUri are set', () => {
        const wrapper = shallow(
            <ProgressiveImage
                {...baseProps}
            />,
        );
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setThumbnail');
        instance.load();

        jest.runAllTimers();
        expect(instance.setThumbnail).toHaveBeenCalledTimes(1);
    });
});