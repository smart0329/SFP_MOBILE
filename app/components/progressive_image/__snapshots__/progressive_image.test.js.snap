// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage should match snapshot 1`] = `
<View
  style={Object {}}
>
  <AnimatedComponent
    blurRadius={5}
    onError={[MockFunction]}
    onLoadEnd={[Function]}
    resizeMethod="auto"
    resizeMode="contain"
    source={
      Object {
        "headers": Object {
          "Authorization": "Bearer ",
        },
        "uri": "https://images.com/image.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
  <AnimatedComponent
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#ffffff",
          "opacity": 0.8,
        },
      ]
    }
  />
</View>
`;
